### PR TITLE
Generate chain address using vault public key

### PR DESCRIPTION
**Pull request by Ehsan-saradar**

### CODE REVIEW

File: web/src/pages/balance/index.tsx [hunk 1]
```diff
@@ -10,8 +10,68 @@ import {
   QRCodeOutlined,
   RefreshOutlined,
 } from "utils/icons";
+import VaultManager from "utils/vault-manager";
+
+interface Chain {
```
Comment (johnnyluo) [inline web/src/pages/balance/index.tsx:15]: Also I think Chain should be just the name or something , here what you really need to display is the `Coin` 

Chain is just the NativeToken on the blockchain

Basically a vault can have many coins , and then you can group the coin based on it's chain for display

File: web/src/utils/vault-manager.ts [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+
+                const address = core.AnyAddress.createWithPublicKey(
+                  publicKey,
+                  core.CoinType.bitcoin
+                ).description();
+
+                resolve(address);
+              })
+              .catch(() => {});
+
```
Comment (johnnyluo) [inline web/src/utils/vault-manager.ts:142]: As you can see this section of code is duplicated many times , so you can actually just write a class that can deal with all the address 
refer to https://github.com/vultisig/vultisig-windows/blob/main/frontend/src/blockchain/coin-factory.ts#L35

File: web/src/pages/balance/index.tsx [hunk 3]
```diff
@@ -10,8 +10,68 @@ import {
...
+  address?: string;
+  icon: string;
+  key: VaultManager.ChainType;
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply web/src/pages/balance/index.tsx]: Fixed

File: web/src/pages/balance/index.tsx (hunk 1 above)
Comment (Ehsan-saradar) [reply web/src/pages/balance/index.tsx]: Updated using the same logic in android and ios

File: web/src/utils/vault-manager.ts [hunk 4]
```diff
@@ -0,0 +1,261 @@
...
+  export interface Derivation {
+    publicKeyEcdsa: string;
+    hexChainCode: string;
+    derivePath: string;
+  }
+
+  let core: WalletCore;
+  let vault: Vault;
+
+  const derivePath = {
```
[the remark this answers is not in the document]
Comment (Ehsan-saradar) [reply web/src/utils/vault-manager.ts]: Removed this part and used wallet-core

File: web/src/utils/vault-manager.ts (hunk 2 above)
Comment (Ehsan-saradar) [reply web/src/utils/vault-manager.ts]: Updated

**Review by johnnyluo — APPROVED** [review]
LGTM

**Pull request merged**